osu.Game.Rulesets.Osu/Difficulty/Preprocessing/OsuDifficultyHitObject.cs

```diff
@@ -71,11 +87,15 @@ private void setDistances()
             {
                 computeSliderCursorPosition(lastSlider);
                 TravelDistance = lastSlider.LazyTravelDistance * scalingFactor;
+                TravelTime = Math.Max(lastSlider.LazyTravelTime / clockRate, min_delta_time);
+                MovementTime = Math.Max(StrainTime - TravelTime, min_delta_time);
+                MovementDistance = Vector2.Subtract(lastSlider.TailCircle.StackedPosition, BaseObject.StackedPosition).Length * scalingFactor;
             }
 
             Vector2 lastCursorPosition = getEndCursorPosition(lastObject);
 
             JumpDistance = (BaseObject.StackedPosition * scalingFactor - lastCursorPosition * scalingFactor).Length;
+            MovementDistance = Math.Min(JumpDistance, MovementDistance);
```

Comment:
These could be set to be equal to straintime for circles. In fact that would probably more accurately match the definition.

Reply:
Yeah that's the thing, it's going against the xmldocs which was unexpected to me at first read.

So basically:
```
MovementDistance = JumpDistance;
MovementTime = StrainTime;
```
For hitcircles?

Reply:
I have no problem with that definition.

Reply:
For all intents and purposes, movement distance could seen as the same thing as junp distance. The idea here in the definition is that there is a leniency of the slider where your movement can be either from the minimum end point of the slider travel distance, or from the actual tail end. This way we can better assess the true observed velocity for the movement to the next note. Perhaps a better way to put it is that when you take into consideration the movement from the slider to the next note, lazy end is not always where its actually easiest to end if the movement after the slider is in the direction of the slider.

Reply:
The slider augmentation here is that we treat sliders as two seperate velocities, rather than 1. The core benefit of viewing things this way is that we buff sliders by assessing two movements instead of one. 

The reason why I added this leniency is because without it, slider jumps of a certain design become overweighted as the secondary velocity to the following note is seen as faster than the actual difficulty in gameplay is. This is due to the inaccurate real cursor position by using lazy position exclusively.

Reply:
Yeah, sure. Again just understand that this will nerf these sort of patterns on the other side of the spectrum:
![image](https://user-images.githubusercontent.com/1329837/139890541-2fcb2af3-1f90-4456-b02d-887bea160ada.png)
(A player will not follow either of those)